File: easy/3917_count_indices_with_opposite_parity/Solution.java
class Solution {
  public int[] countOppositeParity(int[] nums) {
    final int[] parities = new int[nums.length];

    for (int i = 0; i < nums.length; ++i) {
      for (int j = i + 1; j < nums.length; ++j) {
        if (nums[i] % 2 != nums[j] % 2) {
          ++parities[i];
        }
      }
    }

    return parities;
  }
}

File: easy/3917_count_indices_with_opposite_parity/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> countOppositeParity(std::vector<int> &nums) {
    std::vector<int> parities(nums.size(), 0);

    int even = 0;
    int odd = 0;

    for (int i = nums.size() - 1; i > -1; --i) {
      if (nums[i] % 2 == 0) {
        parities[i] = odd;
        ++even;
      } else {
        parities[i] = even;
        ++odd;
      }
    }

    return parities;
  }
};
